src/components/RunGraph.vue

```diff
@@ -4,25 +4,22 @@
 
 <script lang="ts" setup>
   import { onBeforeUnmount, onMounted, ref } from 'vue'
-  import { RunGraphConfig } from '@/models/RunGraph'
-  import { start, stop } from '@/objects'
-  import { RunGraphDomain } from '@/objects/domain'
+  import { RunGraphProps } from '@/models/RunGraph'
+  import { ScaleXDomain, start, stop } from '@/objects'
   import { emitter } from '@/objects/events'
-  // import { WorkerMessage, worker } from '@/workers/runGraph'
 
-  const props = defineProps<{
-    config: RunGraphConfig,
-    domain?: RunGraphDomain,
-  }>()
+  // using the props object as a whole
+  // eslint-disable-next-line vue/no-unused-properties
+  const props = defineProps<RunGraphProps>()
 
   const emit = defineEmits<{
-    (event: 'update:domain', value: RunGraphDomain): void,
+    (event: 'update:viewport', value: ScaleXDomain): void,
   }>()
 
   const stage = ref<HTMLDivElement>()
 
-  emitter.on('domainCreated', domain => emit('update:domain', domain))
-  emitter.on('domainUpdated', domain => emit('update:domain', domain))
+  emitter.on('viewportDateRangeUpdated', range => emit('update:viewport', range))
```

Comment:
Does this not get emitted as soon as the viewport finishes loading? I'm not seeing the associated v-model populated until the viewport changes

Reply:
you might be right, will investigate

Reply:
Fixed in https://github.com/PrefectHQ/graphs/pull/236